### DISCUSSION TITLE
Enable/disable keyboard shortcut

### DISCUSSION THREAD

**Discussion by mcint** [rxhanson/Rectangle#399] [Ideas]
I would like to have a keyboard shortcut to toggle whether Rectangle (via MASShortcut) captures or passes-through (all the other) keyboard shortcuts. I appreciate keyboard shortcuts, so much so that I use this app to add keyboard-controlled window placement and management. However, other applications I would like to control by keyboard, would like to rely on the same key combinations. I hope to get the best of both worlds by modifying Rectangle (in its use of MASShortcut) to stop responding to, and verifying pass-through of, the other shortcuts.

I'm willing and interested to make a PR. I would like to ask, are there gotcha's to be aware of in handling this functionality. I would like to land a simple version of this first, but later, a more extensive version could interact with the "Ignore ApplicationX" toggling, in one of a few ways. I'm [asking in MASShortcut now as well](https://github.com/shpakovski/MASShortcut/issues/157).

**Comment by rxhanson** [comment]
Thanks for looking to contribute! It sounds like you're aware of how currently Rectangle will unbind its shortcuts when the frontmost app is ignored via the menu bar menu.

The simplest form of what you're looking for sounds like just a keyboard shortcut toggle for the app ignore. If that is the case, then I can see merging a pull request that allows this, but I would prefer it to only be configurable via terminal command since I don't have a desire to add another keyboard shortcut recorder to the UI for it.

Beyond that, it's worth noting that MASShortcut uses the Carbon API for registering keyboard shortcuts, and you could hit a wall with what you want out of it since that API effectively binds one shortcut to one application at a time. You would have to change Rectangle to listen to keystrokes rather than use this API if you want to bend this behavior further. I think that a modified version of Rectangle that listens to keystrokes would be better as a fork, but there are a decent amount of users that want functionality that can only be provided by listening to keystrokes.
